### DISCUSSION TITLE
test discussion

### DISCUSSION THREAD

**Discussion by novakovicluka97** [Cyfrin/foundry-full-course-cu#1829] [General]
## heading 2
# heading 3
`
first code
`
I am receiving this error, _here_ is my **code**:
"""
Something wrong

"""

**Comment by novakovicluka97** [comment]
Just testing markdown before I delete this post made by 

code:
'
def sign_message(message, secret_key):
    # Create a new HMAC object using the secret key and SHA256 hashing algorithm
    hmac_obj = hmac.new(secret_key.encode(), message.encode(), hashlib.sha256)

    # Return the signature in hexadecimal format
    return hmac_obj.hexdigest()


# Example usage
message = "Hello, this is a message."
secret_key = "my_secret_key"
signature = sign_message(message, secret_key)
print("Signature:", signature)

print(len("0x0B97Af661CEE77c81144fb2e49E0eDD65aec4574"))
'

> @novakovicluka97 

1. 2131

- 12321

- [ ] task 1

**Comment by novakovicluka97** [comment]
'''
def sign_message(message, secret_key):
    # Create a new HMAC object using the secret key and SHA256 hashing algorithm
    hmac_obj = hmac.new(secret_key.encode(), message.encode(), hashlib.sha256)

    # Return the signature in hexadecimal format
    return hmac_obj.hexdigest()


# Example usage
message = "Hello, this is a message."
secret_key = "my_secret_key"
signature = sign_message(message, secret_key)
print("Signature:", signature)

print(len("0x0B97Af661CEE77c81144fb2e49E0eDD65aec4574"))
'''

**Comment by novakovicluka97** [comment]
`
def sign_message(message, secret_key):
    # Create a new HMAC object using the secret key and SHA256 hashing algorithm
    hmac_obj = hmac.new(secret_key.encode(), message.encode(), hashlib.sha256)

    # Return the signature in hexadecimal format
    return hmac_obj.hexdigest()

# Example usage
message = "Hello, this is a message."
secret_key = "my_secret_key"
signature = sign_message(message, secret_key)
print("Signature:", signature)

print(len("0x0B97Af661CEE77c81144fb2e49E0eDD65aec4574"))
`

**Comment by novakovicluka97** [comment]
```
def sign_message(message, secret_key):
    # Create a new HMAC object using the secret key and SHA256 hashing algorithm
    hmac_obj = hmac.new(secret_key.encode(), message.encode(), hashlib.sha256)

    # Return the signature in hexadecimal format
    return hmac_obj.hexdigest()

# Example usage
message = "Hello, this is a message."
secret_key = "my_secret_key"
signature = sign_message(message, secret_key)
print("Signature:", signature)

print(len("0x0B97Af661CEE77c81144fb2e49E0eDD65aec4574"))
```

**Comment by novakovicluka97** [comment]
'''python
def sign_message(message, secret_key):
    # Create a new HMAC object using the secret key and SHA256 hashing algorithm
    hmac_obj = hmac.new(secret_key.encode(), message.encode(), hashlib.sha256)

    # Return the signature in hexadecimal format
    return hmac_obj.hexdigest()

# Example usage
message = "Hello, this is a message."
secret_key = "my_secret_key"
signature = sign_message(message, secret_key)
print("Signature:", signature)

print(len("0x0B97Af661CEE77c81144fb2e49E0eDD65aec4574"))
'''

**Comment by novakovicluka97** [comment]
```python
def sign_message(message, secret_key):
    # Create a new HMAC object using the secret key and SHA256 hashing algorithm
    hmac_obj = hmac.new(secret_key.encode(), message.encode(), hashlib.sha256)

    # Return the signature in hexadecimal format
    return hmac_obj.hexdigest()

# Example usage
message = "Hello, this is a message."
secret_key = "my_secret_key"
signature = sign_message(message, secret_key)
print("Signature:", signature)

print(len("0x0B97Af661CEE77c81144fb2e49E0eDD65aec4574"))
```

**Reply by EngrPips** [reply]
Nice testing @novakovicluka97.
